drift_dev/lib/src/writer/tables/update_companion_writer.dart

```diff
@@ -194,7 +194,9 @@ class UpdateCompanionWriter {
           ..write(mapSetter)
           ..write('(converter.toSql($getterName.value)');
 
-        if (!column.nullable && scope.generationOptions.nnbd) {
+        if (!column.nullable &&
+            converter.sqlTypeIsNullable &&
+            scope.generationOptions.nnbd) {
```

Comment:
I'm not sure if we need this check at all now. My understanding is that this only happens if we have a non-null value in Dart that may get mapped to SQL `NULL` by the type converter?

In that case, I think we should include the `null` value instead of asserting that it doesn't exist. We may have to change `variableTypeCodeWithoutArray` so that it returns a nullable type if either the column is nullable or if there's a converter with a nullable SQL type (currently it only checks the former).

Reply:
I don't fully understand the second paragraph. This is only eliminates an unnecessary assertion. I think this is not modify the logic at all. Before my change the generator always wrote a `!` at the end of `toSql()` even the TypeConverter has non-nullable type argument (lint: unnecessary_non_null_assertion).